src/mlpack/methods/ann/layer/atrous_convolution_impl.hpp

```diff
@@ -37,6 +37,65 @@ AtrousConvolution<
   // Nothing to do here.
 }
 
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+AtrousConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::AtrousConvolution(
+  const AtrousConvolution& network) :
```

Comment:
This is a bit pedantic, but the name `layer` or `other` might be clearer here.  `network` makes me think we are passing an entire network to copy, but that's not what we're doing. :)